diff --git a/amavisd b/amavisd
index 7f93194..8e3f39d 100755
--- a/amavisd
+++ b/amavisd
@@ -4332,14 +4332,14 @@ sub read_l10n_templates($;$) {
 
 # # attempt to read a list of config files to use instead of the default one,
 # # using an external helper script. Used by the Debian/Ubuntu distribution.
-# sub find_config_files(@) {
-#   my(@dirs) = @_;
-#   local $ENV{PATH} = '/bin:/usr/bin';
-#   my(@config_files) = map { `run-parts --list "$_"` } @dirs;
-#   chomp(@config_files);
-#   # untaint - this data is secure as we check the files themselves later
-#   map { untaint($_) } @config_files;
-# }
+sub find_config_files(@) {
+  my(@dirs) = @_;
+  local $ENV{PATH} = '/bin:/usr/bin';
+  my(@config_files) = map { `run-parts --list "$_"` } @dirs;
+  chomp(@config_files);
+  # untaint - this data is secure as we check the files themselves later
+  map { untaint($_) } @config_files;
+}
 
 #use CDB_File;
 #sub tie_hash($$) {
@@ -19081,10 +19081,10 @@ $Amavis::Conf::map_full_type_to_short_type_re =
 
 # default location of the config file if none specified
 if (!@config_files) {
-  @config_files = ( '/etc/amavisd.conf' );
+# @config_files = ( '/etc/amavisd.conf' );
 # # Debian/Ubuntu specific:
-# @config_files = Amavis::Util::find_config_files('/usr/share/amavis/conf.d',
-#                                                 '/etc/amavis/conf.d');
+  @config_files = Amavis::Util::find_config_files('/usr/share/amavis/conf.d',
+                                                  '/etc/amavis/conf.d');
 }
 
 # Read and evaluate config files, which may override default settings
